### PR TITLE
Add functionality to add imports

### PR DESCRIPTION
**Pull request by roonyh**

### CODE REVIEW

File: modules/web/js/ballerina/components/package-definition.jsx [hunk 1]
```diff
@@ -41,6 +45,22 @@ class PackageDefinition extends React.Component {
         this.props.model.setAttribute('viewState.expanded', !this.props.model.viewState.expanded);
     }
 
+    handleImportsMouseEnter(e) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/package-definition.jsx]: Shall we have doc comments to the functions in this class ? Also we can remove the 'e' variable if not used right ?

File: modules/web/js/ballerina/components/package-definition.jsx [hunk 2]
```diff
@@ -41,6 +45,22 @@ class PackageDefinition extends React.Component {
...
+        this.props.model.setAttribute('viewState.addingImport', true);
+    }
+
+    handleImportsMouseLeave(e) {
+        console.log('ml');
```
Comment (hemikak) [inline modules/web/js/ballerina/components/package-definition.jsx]: Unnecessary log ?

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import Autosuggest from 'react-autosuggest';
+import './suggestions-text.css'
+
+class SuggestionsText extends React.Component {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Shall we have class comments here ?

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+              {suggestion.name}
+            </div>
+        );
+    }
+
+    getSuggestionValue(suggestion) {
+        return suggestion.name;
+    }
+
+    getSuggestions(query) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Shall we have comments for the functions.

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    onKeyDown(e) {
+        if(e.keyCode === 13) {
+            this.props.onEnter(this.state.value);
+            this.setState({
+                value: ''
+            });
+        }
+    }
+
+    onChange(event, { newValue, method }) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Do we need the 'method' argument here ?

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        this.setState({value: newValue});
+    }
+
+    componentWillUnmount() {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Shall we have doc comments here ? 
/**
 \* {@inheritDoc}
 */

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+    componentDidMount() {
+        this.renderSuggestionsText();
+    }
+
+    render() {
+        const {x, y, height, width} = this.props;
+
+        const style = {
+            fill: "#eee"
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Should we move the styles to a separate css file ?

File: modules/web/js/ballerina/components/suggestions-text.jsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        }
+
+        if(!this.props.show) {
+            style.display = 'none'
```
Comment (hemikak) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Ending semi-colon is missing.

File: modules/web/js/ballerina/visitors/dimension-calculator/package-definition-dimension-calculator-visitor.js [hunk 9]
```diff
@@ -49,8 +49,12 @@ class PackageDefinitionDimensionCalculatorVisitor {
             const imports = astRoot.children.filter(
                 c => {return ASTFactory.isImportDeclaration(c)});
 
-            height += topGutter + topBarHeight + importInputHeight +
+            height += topGutter + topBarHeight +
                       imports.length * packageDefinition.importDeclaration.itemHeight;
+
+            if(viewState.addingImport){
+                height += importInputHeight
```
Comment (hemikak) [inline modules/web/js/ballerina/visitors/dimension-calculator/package-definition-dimension-calculator-visitor.js]: Ending semi-colon is missing.

File: modules/web/js/ballerina/components/package-definition.jsx (hunk 1 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/package-definition.jsx]: Added jsdocs for all the methods and fixed lint errors in #2415

File: modules/web/js/ballerina/components/package-definition.jsx (hunk 2 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/package-definition.jsx]: This has been removed in a follow up commit :)

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 3 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Added jsdoc for the class and methods in #2415

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 4 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Added jsdoc for the class and methods in #2415

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 5 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Nope. Removed in #2415. Also fixed other lint issues.

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 6 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Added in #2415

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 7 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Moved to the relevant css file in #2415

File: modules/web/js/ballerina/components/suggestions-text.jsx (hunk 8 above)
Comment (roonyh) [inline modules/web/js/ballerina/components/suggestions-text.jsx]: Fixed in #2415

File: modules/web/js/ballerina/visitors/dimension-calculator/package-definition-dimension-calculator-visitor.js (hunk 9 above)
Comment (roonyh) [inline modules/web/js/ballerina/visitors/dimension-calculator/package-definition-dimension-calculator-visitor.js]: This has been fixed in the process of fixing lint errors.

**Pull request merged**